std/internal/math/biguintcore.d

```diff
@@ -1288,7 +1289,7 @@ BigDigit [] add(const BigDigit [] a, const BigDigit [] b) pure nothrow
 
 /**  return x + y
  */
-BigDigit [] addInt(const BigDigit[] x, ulong y) pure nothrow
+BigDigit [] addInt(const BigDigit[] x, ulong y) pure nothrow @safe
```

Comment:
Simpler would be dumping disassembly, it will clearly show calls to __d_array_boundscheck or some such.
Performance measurements are also possible but it may be the case that it would saturate different bottleneck in bigint implementation.